### ISSUE TITLE
Error on build versin 0.3

### ISSUE THREAD

**Issue by fanningert** [openambitproject/openambit#102]
Currently I get following error when I build the 0.3 version.
Did any known the problem. I don't think it is a problem of the project, but I am not so familiar with cmake.

Build command
```
./build.sh -DCMAKE_BUILD_TYPE=Debug
```

**Update:** I add the complete build output
```
------building libambit------
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
-- checking for module 'libusb-1.0'
--   found libusb-1.0, version 1.0.19
-- Found libusb-1.0: /usr/include/libusb-1.0, /usr/lib64/libusb-1.0.so
-- Found UDev: /usr/include  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/libambit-build
Scanning dependencies of target ambit
[ 12%] Building C object CMakeFiles/ambit.dir/hid.c.o
[ 25%] Building C object CMakeFiles/ambit.dir/protocol.c.o
[ 37%] Building C object CMakeFiles/ambit.dir/debug.c.o
[ 50%] Building C object CMakeFiles/ambit.dir/personal.c.o
[ 62%] Building C object CMakeFiles/ambit.dir/libambit.c.o
[ 75%] Building C object CMakeFiles/ambit.dir/pmem20.c.o
[ 87%] Building C object CMakeFiles/ambit.dir/crc16.c.o
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/pmem20.c: In function ‘libambit_pmem20_gps_orbit_write’:
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/pmem20.c:464:9: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         *((uint32_t*)(&tailbuf[0])) = htole32(PMEM20_GPS_ORBIT_START);
         ^
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/pmem20.c:465:9: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         *((uint32_t*)(&tailbuf[4])) = htole32(bufsizes[0]);
         ^
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/libambit.c: In function ‘libambit_date_time_set’:
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/libambit.c:232:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
     *(uint16_t*)(&date_data[0]) = htole16(1900 + tm->tm_year);
     ^
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/libambit.c:238:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
     *(uint16_t*)(&time_data[0]) = htole16(1900 + tm->tm_year);
     ^
/home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit/libambit.c:243:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
     *(uint16_t*)(&time_data[6]) = htole16(1000*tm->tm_sec);
     ^
[100%] Linking C shared library libambit.so
[100%] Built target ambit
------building openambit------
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for Q_WS_X11
-- Looking for Q_WS_X11 - found
-- Looking for Q_WS_WIN
-- Looking for Q_WS_WIN - not found
-- Looking for Q_WS_QWS
-- Looking for Q_WS_QWS - not found
-- Looking for Q_WS_MAC
-- Looking for Q_WS_MAC - not found
-- Found Qt4: /usr/bin/qmake-qt4 (found version "4.8.7") 
-- Found libambit: /home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/src/libambit, /usr/lib64/libambit.so
-- Found UDev: /usr/include  
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fanningert/Documents/git/aur.archlinux.org/openambit/src/openambit/openambit-build
[  2%] Generating moc_mainwindow.cxx
[  6%] Generating ui_mainwindow.h
[  6%] Generating ui_settingsdialog.h
[  8%] Generating qrc_resources.cxx
[ 12%] Generating moc_settingsdialog.cxx
[ 12%] Generating moc_devicemanager.cxx
[ 14%] Generating movescount/moc_movescountlogchecker.cxx
[ 17%] Generating ui_confirmbetadialog.h
[ 19%] Generating moc_settings.cxx
[ 21%] Generating moc_udevlistener.cxx
[ 23%] Generating moc_confirmbetadialog.cxx
[ 25%] Generating moc_single_application.cxx
[ 27%] Generating moc_logstore.cxx
[ 29%] Generating movescount/moc_movescount.cxx
[ 31%] Generating movescount/moc_movescountjson.cxx
[ 34%] Generating movescount/moc_movescountlogdirentry.cxx
[ 36%] Generating movescount/moc_movescountxml.cxx
Scanning dependencies of target openambit
[ 40%] Building CXX object CMakeFiles/openambit.dir/devicemanager.cpp.o
[ 40%] Building CXX object CMakeFiles/openambit.dir/main.cpp.o
[ 44%] Building CXX object CMakeFiles/openambit.dir/mainwindow.cpp.o
[ 44%] Building CXX object CMakeFiles/openambit.dir/logstore.cpp.o
[ 46%] Building CXX object CMakeFiles/openambit.dir/settingsdialog.cpp.o
[ 48%] Building CXX object CMakeFiles/openambit.dir/settings.cpp.o
[ 51%] Building CXX object CMakeFiles/openambit.dir/logentry.cpp.o
[ 53%] Building CXX object CMakeFiles/openambit.dir/udevlistener.cpp.o
[ 55%] Building CXX object CMakeFiles/openambit.dir/confirmbetadialog.cpp.o
[ 57%] Building CXX object CMakeFiles/openambit.dir/single_application.cpp.o
[ 59%] Building CXX object CMakeFiles/openambit.dir/movescount/movescount.cpp.o
[ 61%] Building CXX object CMakeFiles/openambit.dir/movescount/movescountjson.cpp.o
[ 63%] Building CXX object CMakeFiles/openambit.dir/movescount/movescountlogdirentry.cpp.o
[ 65%] Building CXX object CMakeFiles/openambit.dir/movescount/movescountxml.cpp.o
[ 68%] Building CXX object CMakeFiles/openambit.dir/movescount/movescountlogchecker.cpp.o
[ 70%] Building CXX object CMakeFiles/openambit.dir/qrc_resources.cxx.o
[ 72%] Building CXX object CMakeFiles/openambit.dir/moc_mainwindow.cxx.o
[ 74%] Building CXX object CMakeFiles/openambit.dir/moc_devicemanager.cxx.o
[ 76%] Building CXX object CMakeFiles/openambit.dir/moc_settingsdialog.cxx.o
[ 78%] Building CXX object CMakeFiles/openambit.dir/moc_settings.cxx.o
[ 80%] Building CXX object CMakeFiles/openambit.dir/moc_logstore.cxx.o
[ 82%] Building CXX object CMakeFiles/openambit.dir/moc_udevlistener.cxx.o
[ 85%] Building CXX object CMakeFiles/openambit.dir/moc_confirmbetadialog.cxx.o
[ 87%] Building CXX object CMakeFiles/openambit.dir/moc_single_application.cxx.o
[ 89%] Building CXX object CMakeFiles/openambit.dir/movescount/moc_movescount.cxx.o
[ 91%] Building CXX object CMakeFiles/openambit.dir/movescount/moc_movescountjson.cxx.o
[ 93%] Building CXX object CMakeFiles/openambit.dir/movescount/moc_movescountlogdirentry.cxx.o
[ 95%] Building CXX object CMakeFiles/openambit.dir/movescount/moc_movescountxml.cxx.o
make[2]: *** No rule to make target '/usr/lib64/libambit.so', needed by 'openambit'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 97%] Building CXX object CMakeFiles/openambit.dir/movescount/moc_movescountlogchecker.cxx.o
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/openambit.dir/all' failed
make[1]: *** [CMakeFiles/openambit.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

Regards
Thomas

**Comment by paddy-hack** [comment]
I just ran a build on 0.3 and that completed fine.  Comparing my log with yours, the main difference is that where your log says (abbreviating that really long source code directory a bit):
```
-- Found Qt4: /usr/bin/qmake-qt4 (found version "4.8.7") 
-- Found libambit: /home/fanningert/[...]/openambit/src/openambit/src/libambit, /usr/lib64/libambit.so
-- Found UDev: /usr/include
```
Mine says
```
-- Found Qt4: /usr/bin/qmake (found version "4.8.6") 
-- Found libambit: /code/src/libambit, /code/libambit-build/libambit.so
-- Found UDev: /usr/include  
```
So even though you're building from source, `cmake` finds a `/usr/lib64/libambit.so`.  Do you have `openambit` installed by any chance?
This looks a lot like #77, which was fixed in 1bb38d9, after 0.3 was released.  Please try building the latest revision from git.  That will probably fix this.

**Comment by fanningert** [comment]
@paddy-hack You are right. When I compile 0.3 a different version of openambit is installed.
Yes, I also create a git package and this is working.
https://aur.archlinux.org/packages/openambit-git/
